File: 02_ProgrammingFundamentals/_11_RegularExpressions/_02_Exercise/_01_Furniture.java
package _02_ProgrammingFundamentals._11_RegularExpressions._02_Exercise;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class _01_Furniture {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input value
        String console = scan.nextLine();

        // Add variables
        String regex = ">>(?<furniture>[A-Za-z]+)<<(?<price>[0-9]+.?[0-9]*)!(?<quantity>[0-9]+)";
        Pattern pattern = Pattern.compile(regex);
        List<String> furnitureList = new ArrayList<>();
        double totalMoney = 0;

        // Purchase furniture
        while (!console.equals("Purchase")) {
            Matcher matcher = pattern.matcher(console);
            
            if (matcher.find()) {
                String furniture = matcher.group("furniture");
                
                double price = Double.parseDouble(matcher.group("price"));
                int quantity = Integer.parseInt(matcher.group("quantity"));
                
                furnitureList.add(furniture);
                
                totalMoney += price * quantity;
            }
            
            console = scan.nextLine();
        }
        
        // Print result
        System.out.println("Bought furniture:");
        furnitureList.forEach(System.out::println);
        System.out.printf("Total money spend: %.2f", totalMoney);
        
        scan.close();
    }
}

File: 24_C#Fundamentals/_09_RegularExpressions/_02_Exercise/_01_Furniture/_01_Furniture.cs
using System;
using System.Text.RegularExpressions;

public class _01_Furniture
{
    public static void Main(string[] args)
    {
        // Read input value
        string console = Console.ReadLine();

        // Add variables
        string regex = ">>(?<furniture>[A-Za-z]+)<<(?<price>[0-9]+.?[0-9]*)!(?<quantity>[0-9]+)";
        Regex pattern = new Regex(regex);
        List<string> furnitureList = new List<string>();
        double totalMoney = 0;

        // Purchase furniture
        while (!console.Equals("Purchase"))
        {
            Match matcher = pattern.Match(console);

            if (matcher.Success)
            {
                string furniture = matcher.Groups["furniture"].Value;

                double price = double.Parse(matcher.Groups["price"].Value);
                int quantity = int.Parse(matcher.Groups["quantity"].Value);

                furnitureList.Add(furniture);

                totalMoney += price * quantity;
            }

            console = Console.ReadLine();
        }

        // Print result
        Console.WriteLine("Bought furniture:");
        furnitureList.ForEach(i => Console.WriteLine(i));
        Console.WriteLine($"Total money spend: {totalMoney:F2}");
    }
}
